### ISSUE TITLE
 Performance Issue: Limitation of 500 bytes per request

### ISSUE THREAD

**Issue by jggri** [libplctag/libplctag#506]
Hi,

I'm trying to measure how many tags per second my application can read. During testing, I noticed a limitation in CIP layer to 500 bytes. 
![imagen](https://github.com/user-attachments/assets/a50c3535-afab-44bb-8b18-2d272e9a2436)
Is there any configuration or setting that allows increasing this value?

**Comment by jggri** [comment]
![imagen](https://github.com/user-attachments/assets/a8c31c24-e2cb-43b8-b26c-ff88c30b916b)
![imagen](https://github.com/user-attachments/assets/672ce09d-35fa-4256-af6a-c948dc60263a)

**Comment by kyle-github** [comment]
It depends what kind of PLC and what level of firmware you have.   Newer Rockwell Control and CompactLogix will support the larger sizes (the ethernet module on a ControlLogix also impacts this).

The library tries to use the Forward Open Extended CIP command first.  If that fails, then it drops back to the old Forward Open.  The extended version supports asking for a packet size up to about 4k (8k for some Omron PLCs).   The old Forward Open can only negotiate up to about ~500 bytes.

Note that Micro800 series PLCs cannot negotiate the larger size IIRC.

**Comment by jggri** [comment]
Thanks Kyle. That's what happening
![imagen](https://github.com/user-attachments/assets/9249cb17-807c-4d3b-8121-393260efc8ae)

**Comment by kyle-github** [comment]
Yep.  Looks like that PLC does not support the newer Forward Open.   The old one only has 9 bits for the packet size.  The new one has 16.
